contracts/svm/programs/express_relay/src/lib.rs

```diff
@@ -95,8 +95,8 @@ pub mod express_relay {
         // check "no reentrancy"--submit_bid instruction only used once in transaction
         // this is done to prevent an exploit where a searcher submits a transaction with multiple submit_bid instructions with different permission keys
         // that would allow the searcher to win the right to perform the transaction if they won just one of the auctions
-        let permission_count =
-            num_permissions_in_tx(ctx.accounts.sysvar_instructions.clone(), None)?;
+        let (permission_count, _) =
+            inspect_permissions_in_tx(ctx.accounts.sysvar_instructions.clone(), None)?;
```

Comment:
we can use `get_matching_instructions` here and change `inspect_permissions_in_tx` to something like `calculate_total_router_fee` which gets a list of instructions and calculates the fee.
Right now, it seems that it's doing 2 separate jobs in a single function